cardano-api/src/Cardano/Api/Key.hs

```diff
@@ -67,6 +69,16 @@ generateSigningKey keytype = do
     seedSize = deterministicSigningKeySeedSize keytype
 
 
+generateInsecureSigningKey :: (Key keyrole, SerialiseAsRawBytes (SigningKey keyrole))
+  => AsType keyrole
+  -> IO (SigningKey keyrole)
+generateInsecureSigningKey keytype = do
+  g <- Random.getStdGen
+  let (bs, _) = Random.genByteString (fromIntegral (deterministicSigningKeySeedSize keytype)) g
+  case deserialiseFromRawBytes (AsSigningKey keytype) bs of
+    Just key -> return key
+    Nothing -> error "Unable to generate insecure key"
```

Comment:
A better error message would be `"generateInsecureSigningKey: Unable to generate insecure key"`